qa327/frontend.py

```diff
@@ -109,6 +109,8 @@ def wrapped_inner():
                 # if the user exists, call the inner_function
                 # with user as parameter
                 return inner_function(user)
+            else:
+                return redirect('/login')
         else:
             # else, redirect to the login page
             return redirect('/login')
```

Comment:
Don't think the `else` statements and extra `redirect` are necessary. You can achieve the same with
```
if 'logged_in' in session:
  # assign variables

  if user:
    return inner_function(user)

return redirect('/login')
```
If we are logged in and there is no user, we would never enter the `if user` condition. We would correctly proceed to `redirect` to login.

If we're not logged in, we wouldn't enter the `if logged_in` condition. Likewise, we correctly proceed to `redirect` as well.

The one `redirect` line i propose would be sufficient to achieve what we want. And makes the code more readable.